packages/multi-select-combo-box/src/vaadin-multi-select-combo-box-internal.js

```diff
@@ -246,6 +259,52 @@ class MultiSelectComboBoxInternal extends ComboBoxDataProviderMixin(ComboBoxMixi
       );
     }
   }
+
+  /**
+   * Override method inherited from the combo-box
+   * to render only selected items when read-only,
+   * even if a different set of items is provided.
+   *
+   * @protected
+   * @override
+   */
+  _setOverlayItems(items) {
+    const effectiveItems = this.readonly ? this.selectedItems : items;
+    super._setOverlayItems(effectiveItems);
+  }
+
+  /**
+   * Override method inherited from the combo-box
+   * to not request data provider when read-only.
+   *
+   * @param {number}
+   * @return {boolean}
+   * @protected
+   * @override
+   */
+  _shouldLoadPage(page) {
+    if (this.readonly) {
+      return false;
+    }
+
+    return super._shouldLoadPage(page);
+  }
+
+  /** @private */
+  _readonlyItemsChanged(readonly, selectedItems) {
+    if (readonly && selectedItems) {
+      this.__savedItems = this._getOverlayItems();
+      this._setOverlayItems(selectedItems);
+    }
+
+    // Restore the original dropdown items
+    if (this._oldReadOnly && this.__savedItems) {
```

Comment:
`_oldReadOnly` wouldn't be needed if we check `readonly === false` here instead?

Reply:
Not a blocker for this PR. Just a nit.

Reply:
Thanks, updated the PR accordingly.